cmp/grid/columns/Column.js

```diff
@@ -380,8 +380,11 @@ export class Column {
                 getReactContainerClasses() {return ['xh-grid-tooltip']}
                 render() {
                     const agParams = this.props,
-                        {value, rowIndex, api} = agParams,
-                        record = api.getDisplayedRowAtIndex(rowIndex).data;
+                        {rowIndex, api, colDef} = agParams,
+                        {headerTooltip} = colDef,
+                        value = isNil(rowIndex) ? headerTooltip : agParams.value;
+
+                    const record = api.getDisplayedRowAtIndex(rowIndex)?.data;
 
                     // ag-Grid encodes the value, so we decode it before passing to the renderer
                     return me.tooltipElement(decodeURIComponent(value), {record, column: me, gridModel, agParams});
```

Comment:
I think this could be problematic in that these renderers would be written by the developer to expect the cell contents - e.g. a number to format - and now they would be unexpectedly getting the `headerTooltip` string.

I think we would need to short-circuit the call to `me.tooltipElement` in this case and return the `headerTooltip` directly - probably needs to be wrapped in some minimal element...  

Am I correct, would this be a potential real-world problem?

Reply:
Right, I thought of this also, but the AGGrid docs say that a tooltipComponent needs to ready to handle both potentials.

If the tooltipComponent returns just the headerTooltip string, it gets rendered rather plainly, not in the nice AG-Grid tooltip box.  So our TooltipComponent would need to style the header tooltip itself.  Which defeats the purpose of a tooltipElement (which I guess  is to give the dev total control over look of the tooltip).  Maybe it needs to be made more clear in the docs that a tooltipElement also needs to be ready to style a tooltipHeader if a tooltipHeader is on the column - that is essentially what AG-Grid is saying.

The tooltipElement api could be made more specific to accept 
`me.tooltipElement({cellValue: decodeURIComponent(value), headerValue: headerTooltip}, {record,....`
That might make it more clear as well.

Reply:
My thinking there is that `Column.tooltipElement` is intended to show rich tooltips for the grid data itself, and the fact that it gets called when hovering over the header is honestly somewhat weird - we have the simpler `headerTooltip` string config for that. 

So if we handled the hover-over-header case to return a reasonable-looking (but determined by Hoist) wrapper around any configured `headerTooltip` text I don't think it would violate developer expectations. We could also clarify in our docs that the non-header tip configs are only applied to data.

I don't think that I would want to change the signature of `tooltipElementFn` because it lines up with all the other tip and renderer fns we support where a value is passed as the first arg (and would be a breaking change).

I dunno - would like someone else to weigh in on it, also we have #2006 outstanding and it seems like that's in the same part of the code.

(@haynesjm42 or @lbwexler if you have any time to look at this change I would appreciate your input.)

Reply:
Ok I can see that argument, that `Column.tooltipElement` is a Hoist optimization for rich data tooltips.   So, if a developer really needs a rich header tooltip, he can not use `Column.tooltipElement` and instead use the escape hatch 'agOptions.tooltipComponentFramework'.   

I'll go ahead and create a nicely styled header tooltip in the Column.js' `tooltipComponentFramework`.

Reply:
Hi,  I've reworked `column.tooltipComponentFramework` so that if it is called for a headerTooltip it returns a div that will be styled by agGrid tooltip classes.   

I also found a prop `location` that seems to be a better, clearer choice for detecting if overHeader or not.